File: frontend/src/types/recipientAddress.ts
import type {Address} from "./address.ts";

export interface RecipientSuggestionResponse {
    address: Address;
    type: RecipientAddressType;
    recipientStats: RecipientStats;
}

export interface RecipientAddress {
    address: Address;
    recipientStats: RecipientStats;
}

export interface RecipientAddressSuggestionRequest {
    phone: number;
}

export interface RecipientAddressRequest {
    id?: number;
    wardCode: number;
    wardName: string;
    cityCode: number;
    cityName: string;
    detail: string;
    name: string;
    phoneNumber: string;
    latitude?: number;
    longitude?: number;
}

export interface RecipientStats {
    totalSystemOrders: number;
    successRate: number;
    returnedRate: number;
    latestOrderDate: string;
}

export type RecipientAddressType = 'NONE' | 'HISTORY' | 'SAVED';

File: backend/src/main/java/com/logistics/response/user/recipientaddress/RecipientAddress.java
package com.logistics.response.user.recipientaddress;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class RecipientAddress {
    private Integer id;
    private String name;
    private String phoneNumber;
    private String fullAddress;
    private Integer cityCode;
    private String cityName;
    private Integer wardCode;
    private String wardName;
    private String detail;
    private Double latitude;
    private Double longitude;
}